replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/modern/RepSyncImpl.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.modern;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashMap;
+import java.util.Map;
+import org.codice.ditto.replication.api.modern.RepSync;
+
+/**
+ * RepSyncImpl represents a RepSync and has methods that allow it to easily be converted to, or
+ * from, a map.
+ */
+public class RepSyncImpl extends AbstractPersistable implements RepSync {

```

Comment:
❓ This is going to be a little difficult from the GraphQL API since now we have the configuration separated between `RepSync` and `ReplicatorConfig`. I'm wondering if we should just combine them now?